drivers/spi/spi_mcux_flexio.c

```diff
@@ -0,0 +1,447 @@
+	mcux_flexio_irq_disable(config->flexio_dev);
+
+	spi_mcux_transfer_next_packet(dev);
+
+	mcux_flexio_irq_enable(config->flexio_dev);
+	mcux_flexio_unlock(config->flexio_dev);
+
+	ret = spi_context_wait_for_completion(&data->ctx);
+out:
+	spi_context_release(&data->ctx, ret);
+
+	return ret;
+}
+
+static int spi_mcux_transceive(const struct device *dev,
+				const struct spi_config *spi_cfg,
+				const struct spi_buf_set *tx_bufs,
+				const struct spi_buf_set *rx_bufs)
+{
+	return transceive(dev, spi_cfg, tx_bufs, rx_bufs, false, NULL, NULL);
+}
+
+#ifdef CONFIG_SPI_ASYNC
+static int spi_mcux_transceive_async(const struct device *dev,
+				const struct spi_config *spi_cfg,
+				const struct spi_buf_set *tx_bufs,
+				const struct spi_buf_set *rx_bufs,
+				spi_callback_t cb,
+				void *userdata)
+{
+	return transceive(dev, spi_cfg, tx_bufs, rx_bufs, true, cb, userdata);
+}
+#endif /* CONFIG_SPI_ASYNC */
+
+static int spi_mcux_release(const struct device *dev,
+				const struct spi_config *spi_cfg)
+{
+	struct spi_mcux_flexio_data *data = dev->data;
+
+	spi_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+static int spi_mcux_init(const struct device *dev)
+{
+	const struct spi_mcux_flexio_config *config = dev->config;
+	struct spi_mcux_flexio_data *data = dev->data;
+	int err;
+
+	err = spi_context_cs_configure_all(&data->ctx);
+	if (err < 0) {
+		return err;
+	}
+
+	spi_context_unlock_unconditionally(&data->ctx);
+
+	data->dev = dev;
+
+	/* TODO: DMA */
+
+#ifdef CONFIG_PINCTRL
+	err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+	if (err) {
+		return err;
+	}
+#endif /* CONFIG_PINCTRL */
+
+	spi_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+static const struct spi_driver_api spi_mcux_driver_api = {
+	.transceive = spi_mcux_transceive,
+#ifdef CONFIG_SPI_ASYNC
+	.transceive_async = spi_mcux_transceive_async,
+#endif
+	.release = spi_mcux_release,
+};
+
+#ifdef CONFIG_PINCTRL
+#define SPI_MCUX_FLEXIO_SPI_PINCTRL_DEFINE(n) PINCTRL_DT_INST_DEFINE(n);
+#define SPI_MCUX_FLEXIO_SPI_PINCTRL_INIT(n) .pincfg = PINCTRL_DT_INST_DEV_CONFIG_GET(n),
+#else
+#define SPI_MCUX_FLEXIO_SPI_PINCTRL_DEFINE(n)
+#define SPI_MCUX_FLEXIO_SPI_PINCTRL_INIT(n)
+#endif /* CONFIG_PINCTRL */
+
+
+#define SPI_MCUX_FLEXIO_SPI_INIT(n)					\
+	SPI_MCUX_FLEXIO_SPI_PINCTRL_DEFINE(n)				\
+									\
+	static FLEXIO_SPI_Type flexio_spi_##n = {			\
+		.flexioBase = (FLEXIO_Type *)DT_REG_ADDR(DT_INST_PARENT(n)), \
+		.SDOPinIndex = DT_INST_PROP(n, sdo_pin),		\
+		.SDIPinIndex = DT_INST_PROP(n, sdi_pin),		\
+		.SCKPinIndex = DT_INST_PROP(n, sck_pin),		\
+	};								\
+									\
+	static const struct spi_mcux_flexio_config spi_mcux_flexio_config_##n = { \
+		.flexio_spi = &flexio_spi_##n,				\
+		.flexio_dev = DEVICE_DT_GET(DT_INST_PARENT(n)),		\
+		SPI_MCUX_FLEXIO_SPI_PINCTRL_INIT(n)			\
+	};								\
+									\
+	static const STRUCT_SECTION_ITERABLE(mcux_flexio_child, mcux_flexio_child_##n) = { \
```

Comment:
Should we use a unique name such as `mcux_flexio_spi_child_##n` instead of `mcux_flexio_child_##n`